### ISSUE TITLE
Action mailbox migration does not respect the active record's primary_key_type

### ISSUE THREAD

**Issue by ExMember** [rails/rails#44289]
### Steps to reproduce

Configure your app to generate tables with UUID primary keys:
config/application.rb:

```ruby
  config.generators do |g|
    g.orm :active_record, primary_key_type: :uuid
  end
```

Run the ActiveMailer installer and migrations
```sh
bin/rails action_mailbox:install
bin/rails db:migrate
```

### Expected behavior

The `action_mailbox_inbound_emails` should have ids with type UUID.

Creating emails fails with the error: `null value in column "record_id" of relation "active_storage_attachments"'

### Actual behavior

The `action_mailbox_inbound_emails` should have ids with type integer.
 
### System configuration
**Rails version**: 7.0.1

**Ruby version**: 3.0.3p157

**Comment by ExMember** [comment]
The active storage migrations had the same problem, reported in #40986.

The most ham-handed fix is to use insert code from that migration into the `CreateActionMailboxTables` migration.

```ruby
# This migration comes from action_mailbox (originally 20180917164000)
class CreateActionMailboxTables < ActiveRecord::Migration[6.0]
  def change
    # Use Active Record's configured type for primary and foreign keys
    primary_key_type, foreign_key_type = primary_and_foreign_key_types

    create_table :action_mailbox_inbound_emails, id: primary_key_type do |t|
      t.integer :status, default: 0, null: false
      t.string  :message_id, null: false
      t.string  :message_checksum, null: false

      t.timestamps

      t.index %i[message_id message_checksum],
        name: 'index_action_mailbox_inbound_emails_uniqueness', unique: true
    end
  end

  private

  def primary_and_foreign_key_types
    config = Rails.configuration.generators
    setting = config.options[config.orm][:primary_key_type]
    primary_key_type = setting || :primary_key
    foreign_key_type = setting || :bigint
    [primary_key_type, foreign_key_type]
  end
end
```

**Comment by CodingByHarry** [comment]
Confirming this is still a problem on Rails 7.0.4.

**Comment by skipkayhil** [comment]
> Confirming this is still a problem on Rails 7.0.4.

Thanks for bumping this, I've opened a PR.

**Comment by skipkayhil** [comment]
Fixed in #48126
